### PR TITLE
Refactor program to module

### PR DESCRIPTION
**Pull request by scxiao**

### CODE REVIEW

**Review by pfultz2 — APPROVED** [review]

**Comment by pfultz2** [comment]
This can be merged in after we merge in PR #680.

**Review by kahmed10 — APPROVED** [review]

**Comment by pfultz2** [comment]
This can be merged now, its passing.

**Pull request merged**
